hooks/useFetch.js

```diff
@@ -141,30 +146,26 @@ const useFetchMany = (urls) => {
   }, [key])
 
   useEffect(() => {
-    const base = from(urls).pipe(
-      mergeMap((url) => from(Object.entries(fetchData(url)))),
-    )
-
-    const errors = new Subject().pipe(
-      filter(isError),
-      pluck('error'),
-      updateBy('url'),
-    )
+    // only take one for no auto-refresh
+    let base = from(urls).pipe(map((url) => fetchData(url).pipe(take(1))))
+    if (refresh) {
+      base = from(urls).pipe(map((url) => fetchData(url)))
+    }
 
-    const sub = from(base)
+    const sub = base
       .pipe(
-        filter(([stream]) => stream === 'data'),
-        pluck(1),
-        mergeAll(),
-        tap(errors),
-        // This is dumb and AFAICT the most elegant way to silently drop
-        // errors.
-        filter((i) => !isError(i)),
-        dematerialize(),
-        updateBy('url'),
+        combineAll(),
+        tap((vals) => {
+          const errs = _map(
+            _filter(vals, (val) => val.kind === 'E' && val.error),
+            // eslint-disable-next-line lodash/prop-shorthand
+            (err) => err.error,
+          )
+          setError(errs)
+          setLoading(false)
+        }),
       )
       .subscribe(setData)
```

Comment:
Won't `setData` end up adding `Notification` including errors and completes?

Reply:
🤷‍♂️ , i have to admit that I refactored the rxjs plumping without fully getting into what the playloads look like or reasoning about it. My 1-2h best quick effort. - Should be easy to modify this to your expectations here.